api/src/main/java/io/jsonwebtoken/security/KeyGenerator.java

```diff
@@ -0,0 +1,30 @@
+package io.jsonwebtoken.security;
+
+import io.jsonwebtoken.SignatureAlgorithm;
+
+import javax.crypto.SecretKey;
+
+/**
+ * Implementations of KeyGenerator will generate {@link SecretKey} based for a given {@link SignatureAlgorithm}.
+ *
+ * @since 0.11.0
+ */
+public interface KeyGenerator {
```

Comment:
I'm not sure about this interface or `KeyPairGenerator`.  In the very new future (1.0), SignatureAlgorithm will be an interface, not an enum, to allow for custom signature algorithm implementations.

Based on this, I'm wondering if the new `SignatureAlgorithm` implementation should implement new interfaces, `SecretKeyAlgorithm` and `KeyPairAlgorithm` and those interfaces have `generateKey` and `generateKeyPair` methods respectively.

Anyway, is there any harm in retaining the current runtime classpath lookup behavior until we can migrate the enums to an interface?